tests/reference.rs

```diff
@@ -7,23 +7,31 @@ use std::str::FromStr;
 
 #[test]
 fn valid_references() {
-    let tcases = vec![
-        "library/busybox",
-        "busybox",
-        "busybox:tag",
-        "busybox:5000",
-        "busybox@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
-        "registry.example.com:5000/library/busybox:5000",
-    ];
-
-    for t in tcases {
-        let r = Reference::from_str(t);
-        asserting(t).that(&r).is_ok();
-        let repo = r.unwrap().repository();
-        asserting(t)
-            .that(&repo.as_str())
-            .is_equal_to("library/busybox");
+    fn run_testcases(tcases: &[&str], expected: &str) {
+        for t in tcases {
+            let r = Reference::from_str(t);
+            asserting(t).that(&r).is_ok();
+            let repo = r.unwrap().repository();
+            asserting(t).that(&repo.as_str()).is_equal_to(expected);
+        }
     }
+    run_testcases(
+        &[
```

Comment:
Can you please split these testcases further in two blocks, so that we check the first half gets the default registry while the second half gets custom values?

Reply:
I thought that's exactly what I did, isn't that so?

Reply:
Oh, registry, I assumed repository.